src/loaders/express/tests/helmet.spec.ts

```diff
@@ -0,0 +1,162 @@
+      'https://www.google.com/recaptcha/',
+      'https://www.recaptcha.net/recaptcha/',
+      'https://www.gstatic.com/recaptcha/',
+      'https://www.gstatic.cn/',
+      'https://www.google-analytics.com/',
+    ],
+    connectSrc: [
+      "'self'",
+      'https://www.google-analytics.com/',
+      'https://ssl.google-analytics.com/',
+      'https://sentry.io/api/',
+      config.aws.attachmentBucketUrl,
+      config.aws.imageBucketUrl,
+      config.aws.logoBucketUrl,
+    ],
+    frameSrc: [
+      "'self'",
+      'https://www.google.com/recaptcha/',
+      'https://www.recaptcha.net/recaptcha/',
+    ],
+    objectSrc: ["'none'"],
+    styleSrc: [
+      "'self'",
+      'https://www.google.com/recaptcha/',
+      'https://www.recaptcha.net/recaptcha/',
+      'https://www.gstatic.com/recaptcha/',
+      'https://www.gstatic.cn/',
+      // For inline styles in angular-sanitize.js
+      "'sha256-b3IrgBVvuKx/Q3tmAi79fnf6AFClibrz/0S5x1ghdGU='",
+    ],
+    formAction: ["'self'"],
+  }
+
+  beforeAll(() => {
+    mockHelmet.xssFilter = jest.fn().mockReturnValue('xssFilter')
+    mockHelmet.noSniff = jest.fn().mockReturnValue('noSniff')
+    mockHelmet.ieNoOpen = jest.fn().mockReturnValue('ieNoOpen')
+    mockHelmet.dnsPrefetchControl = jest
+      .fn()
+      .mockReturnValue('dnsPrefetchControl')
+    mockHelmet.hidePoweredBy = jest.fn().mockReturnValue('hidePoweredBy')
+    mockHelmet.referrerPolicy = jest.fn().mockReturnValue('referrerPolicy')
+    mockHelmet.contentSecurityPolicy = jest
+      .fn()
+      .mockReturnValue('contentSecurityPolicy')
+    mockHelmet.hsts = jest
+      .fn()
+      .mockReturnValue(jest.fn().mockReturnValue('hsts'))
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should call the correct helmet functions', () => {
+    helmetMiddlewares()
+    expect(mockHelmet.xssFilter).toHaveBeenCalled()
+    expect(mockHelmet.noSniff).toHaveBeenCalled()
+    expect(mockHelmet.ieNoOpen).toHaveBeenCalled()
+    expect(mockHelmet.dnsPrefetchControl).toHaveBeenCalled()
+    expect(mockHelmet.hidePoweredBy).toHaveBeenCalled()
+    expect(mockHelmet.referrerPolicy).toHaveBeenCalled()
+    expect(mockHelmet.contentSecurityPolicy).toHaveBeenCalled()
+  })
+
+  it('should call helmet.hsts() if req.secure', () => {
+    const mockReq = expressHandler.mockRequest({ secure: true })
+    const mockRes = expressHandler.mockResponse()
+
+    helmetMiddlewares().filter((result) => typeof result === 'function')[0](
+      mockReq,
+      mockRes,
+      jest.fn(),
+    )
+    expect(mockHelmet.hsts).toHaveBeenCalled()
+  })
+
+  it('should not call helmet.hsts() if !req.secure', () => {
+    const mockReq = expressHandler.mockRequest({ secure: false })
+    const mockRes = expressHandler.mockResponse()
+
+    helmetMiddlewares().filter((result) => typeof result === 'function')[0](
+      mockReq,
+      mockRes,
+      jest.fn(),
+    )
+    expect(mockHelmet.hsts).not.toHaveBeenCalled()
```

Comment:
I'd suggest setting a spy on `next` and making sure the spy gets called

Reply:
edited